src/components/MainPage.js

```diff
@@ -135,34 +154,49 @@ class MainPage extends Component {
   		the state is toggled to show the other view.  Main view is for the comparing
   		currencies.  the other view is to see the supported currencies.
    */
-	handleShowCurrencyList = () => {
-		this.setState({showingCurrencyList: true})
-	}
-	handleHideCurrencyList = () => {
-		this.setState({showingCurrencyList: false})
-	}
+  /*
+  		I have removed the two individual methods to then just make one method which can
+  		toggle the "Supported Currencies" view.  I am doing this so I can have the button
+  		show a 'go back' option to create a better UX as right now its not clear how to get
+  		back to the main app view.
+   */
+
+	handleToggleCurrencyView = () => {
+		this.setState({showingCurrencyList: !this.state.showingCurrencyList});
+		/*
+			Here I added this to refresh the state when user returns to main view from the
+			Supported Currencies page.
+		*/
+		this.state.showingCurrencyList === true ?
+			this.setState({
+				selectedCrypto: this.state.cryptoList[0].code,
+				selectedIntl: this.state.intlList[0].code,
+			}) : null
+	};
 	
   render() {
     return (
       <div>
 				<Header 
-					handleShowCurrencyList={this.handleShowCurrencyList}
-					handleHideCurrencyList={this.handleHideCurrencyList}
+					handleToggleCurrencyView={this.handleToggleCurrencyView}
+					showingCurrencyList={this.state.showingCurrencyList}
```

Comment:
Removing the old methods `handleShowCurrencyList` & `handleHideCurrencyList` since they were combined into the `handleToggleCurrencyView` method.